src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,30 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        lock.readLock().lock();
+        try {
+            list.get(index);
+        } finally {
+            lock.readLock().unlock();
+        }
+        return list.get(index);
```

Comment:
The `get` method accesses the list element twice: once in line 24 and again in line 28. This is inefficient and could lead to inconsistent results if the list is modified between these accesses. You should store the result of `list.get(index)` in a variable and return that variable.